### PR TITLE
feat: DORIS-2046: [tvOS] reuse native part of react-native-dice-video…

### PR DESCRIPTION
**Pull request by NickWang618**
add react-native-dice-video as dependence for react-native-video

### CODE REVIEW

File: ios/Video/NewPlayerView.swift [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+    }
+    
+    func replaceAdTagParameters(payload: NSDictionary) {
+        if let payload = payload as? Dictionary<String, Any> {
+            jsPlayerView?.replaceAdTagParameters(adTagParameters: payload, validFrom: nil, validUntil: nil)
+        }
+    }
+    
+    func convertRNVideoJSPropsToRNDV() -> RNDReactNativeDiceVideo.JSProps {
+        let rndvJsProps = RNDReactNativeDiceVideo.JSProps()
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: can you move this method to a separate file please as a static func of some new props mapper class if possible split this into multiple methods

File: ios/Video/NewPlayerView.swift [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+
+    private func setupDoris() {
+        if let jsBridge = self.jsBridge {
+            let jsProbs = self.convertRNVideoJSPropsToRNDV()
+            var jsPlayerView = RNDReactNativeDiceVideo.JSPlayerView(overlayBuilder: JSOverlayBuilder(bridge: jsBridge), jsProps: jsProbs)
+            self.addSubview(jsPlayerView)
+            jsPlayerView.setRCTBubblingEventBlock(onVideoProgress: self.onVideoProgress,
+                                                  onBackButton: self.onBackButton,
+                                                  onVideoError: self.onVideoError,
+                                                  onRequestPlayNextSource: nil,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: onRequestPlayNextSource: onRelatedVideoClicked

File: ios/Video/NewPlayerView.swift [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+                subPropertyId: sourceValue.config.muxData.subPropertyId,
+                videoIsLive: sourceValue.config.muxData.videoIsLive,
+                experimentName: sourceValue.config.muxData.experimentName,
+                videoSeries: nil,
+                videoCdn: nil,
+                videoDuration: nil)
+            let jsBeacon = JSConfig.JSBeacon(
+                authUrl: sourceValue.config.beacon?.authUrl.absoluteString,
+                url: sourceValue.config.beacon?.url.absoluteString,
+                headers: nil)
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: pass nil for beacons, they are handled on JS side for TV

File: ios/Video/NewPlayerView.swift [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+                progressUpdateInterval: sourceValue.progressUpdateInterval ?? 6,
+                type: sourceValue.type,
+                title: sourceValue.title ?? "",
+                live: sourceValue.live,
+                partialVideoInformation: jsPartialVideoInformation,
+                isAudioOnly: sourceValue.isAudioOnly,
+                config: jsConfig,
+                imageUri: sourceValue.imageUri,
+                thumbnailsPreview: sourceValue.thumbnailsPreview,
+                resumePosition: nil,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: resumePositionValue should be from here

func setInitialSeek(position: Double) {
        jsProps.startAt.value = position
}

File: ios/Video/NewPlayerView.swift [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+                delay: nil,
+                ads: jsAds,
+                subtitles: jsSubtitles,
+                limitedSeekableRange: jsLimitedSeekableRange,
+                selectedAudioTrack: nil,
+                selectedSubtitleTrack: sourceValue.selectedSubtitleTrack,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: need to add **preferredAudioTracks** property it is missing on **rndv** side, and copy logic from rnv to rndv

File: ios/Video/NewPlayerView.swift [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+                selectedPlaybackQuality: nil,
+                nowPlaying: jsNowPlaying)
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: should be **trackPolicy** prop as well looks like it is missing also

File: ios/Video/NewPlayerView.swift [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+            dorisTranslationsViewModel.fastForward = translationsValue.fastForward
+            dorisTranslationsViewModel.audio = translationsValue.audioTracks
+            dorisTranslationsViewModel.info = translationsValue.info
+            dorisTranslationsViewModel.adsCountdownAd = translationsValue.adsCountdownAd
+            dorisTranslationsViewModel.adsCountdownOf = translationsValue.adsCountdownOf
+            dorisTranslationsViewModel.annotations = translationsValue.annotations
+            dorisTranslationsViewModel.playingLive = translationsValue.playingLive
+            dorisTranslationsViewModel.nowPlaying = translationsValue.nowPlaying
+//            dorisTranslationsViewModel.tvPlayerEPG = translationsValue.tvPlayerEPG
+            jsTranslations = RNDReactNativeDiceVideo.JSTranslations(beaconTranslations: nil, dorisTranslations: dorisTranslationsViewModel)
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: what are those commented ones?

File: ios/Video/NewPlayerView.swift [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+        if let themeValue = self.jsProps.theme.value {
+            let fonts = JSTheme.JSFonts(
+                secondaryFontName: themeValue.fonts.secondary,
+                primaryFontName: themeValue.fonts.primary,
+                tertiaryFontName: "")
+            //need confirm
+            let colors = JSTheme.JSColors(
+                accentColor: themeValue.colors.primary,
+                backgroundColor: nil)
+            jsTheme = JSTheme(fonts: fonts, colors: colors)
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: looks fine

File: ios/Video/NewPlayerView.swift [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+            return RNDReactNativeDiceVideo.JSTrackPolicyPair(audio: trackPolicyPair.audio, subtitle: trackPolicyPair.subtitle)
+        }) {
+            jsTracksPolicy = RNDReactNativeDiceVideo.JSTracksPolicy(items: items)
+        }
+        
+        let rndvJSVideoDataConfig = RNDReactNativeDiceVideo.JSVideoData.JSVideoDataConfig(
+            translations: jsTranslations,
+            buttons: jsButtons,
+            theme: jsTheme,
+            playlist: nil,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: playlist should be mapped from **relatedVideos** prop

File: ios/Video/NewPlayerView.swift [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+            testIdentifiers: nil,
+            annotations: nil,
+            overlayConfig: jsOverlayConfig,
+            tracksPolicy: jsTracksPolicy,
+            isFullScreen: self.isFullScreen,
+            allowAirplay: self.allowAirplay,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: set false here

File: ios/Video/NewPlayerView.swift [hunk 11]
```diff
@@ -0,0 +1,394 @@
...
+        let rndvJSVideoDataConfig = RNDReactNativeDiceVideo.JSVideoData.JSVideoDataConfig(
+            translations: jsTranslations,
+            buttons: jsButtons,
+            theme: jsTheme,
+            playlist: nil,
+            testIdentifiers: nil,
+            annotations: nil,
+            overlayConfig: jsOverlayConfig,
+            tracksPolicy: jsTracksPolicy,
+            isFullScreen: self.isFullScreen,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: set true here

File: ios/Video/NewPlayerView.swift [hunk 12]
```diff
@@ -0,0 +1,394 @@
...
+            allowAirplay: self.allowAirplay,
+            canMinimise: self.canMinimise,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: set false here

File: ios/Video/NewPlayerView.swift [hunk 13]
```diff
@@ -0,0 +1,394 @@
...
+            isPipEnabled: nil,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: false

File: ios/Video/NewPlayerView.swift [hunk 14]
```diff
@@ -0,0 +1,394 @@
...
+            canShareplay: nil,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: false

File: ios/Video/NewPlayerView.swift [hunk 15]
```diff
@@ -0,0 +1,394 @@
...
+            testIdentifiers: nil,
+            annotations: nil,
+            overlayConfig: jsOverlayConfig,
+            tracksPolicy: jsTracksPolicy,
+            isFullScreen: self.isFullScreen,
+            allowAirplay: self.allowAirplay,
+            canMinimise: self.canMinimise,
+            isPipEnabled: nil,
+            canShareplay: nil, 
+            isPlaybackQualityChangeAllowed: nil,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: false

File: ios/Video/NewPlayerView.swift [hunk 16]
```diff
@@ -0,0 +1,394 @@
...
+            var jsPlayerView = RNDReactNativeDiceVideo.JSPlayerView(overlayBuilder: JSOverlayBuilder(bridge: jsBridge), jsProps: jsProbs)
+            self.addSubview(jsPlayerView)
+            jsPlayerView.setRCTBubblingEventBlock(onVideoProgress: self.onVideoProgress,
+                                                  onBackButton: self.onBackButton,
+                                                  onVideoError: self.onVideoError,
+                                                  onRequestPlayNextSource: nil,
+                                                  onFullScreenButton: nil,
+                                                  onVideoStart: self.onVideoLoadStart,
+                                                  onVideoEnded: self.onVideoEnd,
+                                                  onVideoPaused: nil,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: need to map **onVideoPaused** from rndv into **onPlaybackRateChange** of rnv

File: ios/Video/NewPlayerView.swift (hunk 10 above)
Comment (NickWang618) [reply ios/Video/NewPlayerView.swift:322]: how about canMinimise

File: ios/Video/NewPlayerView.swift (hunk 7 above)
Comment (NickWang618) [reply ios/Video/NewPlayerView.swift]: need confirm Translations have those value, but JSTranslations not have

File: ios/Video/NewPlayerView.swift (hunk 5 above)
Comment (NickWang618) [reply ios/Video/NewPlayerView.swift]: selectedAudioTrack is a String,  preferredAudioTracks is a [String], should I use preferredAudioTracks[0]?

File: ios/Video/NewPlayerView.swift [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+            var jsPlayerView = RNDReactNativeDiceVideo.JSPlayerView(overlayBuilder: JSOverlayBuilder(bridge: jsBridge), jsProps: jsProbs)
+            self.addSubview(jsPlayerView)
+            jsPlayerView.setRCTBubblingEventBlock(onVideoProgress: self.onVideoProgress,
+                                                  onBackButton: self.onBackButton,
+                                                  onVideoError: self.onVideoError,
+                                                  onRequestPlayNextSource: self.onRelatedVideoClicked,
+                                                  onFullScreenButton: nil,
+                                                  onVideoStart: self.onVideoLoadStart,
+                                                  onVideoEnded: self.onVideoEnd,
+                                                  onVideoPaused: self.onPlaybackRateChange,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: please check the payload sent inside this two callbacks separatelly on rnv and rndv they should contain all expected props in this case for example 
when **onVideoPaused** is called from rndv its payload looks like this
{"isPaused": true} or {"isPaused": false}
but rnv expects it like this on JS side
{"playbackRate": 1.0} or {"playbackRate": 0.0}
to make it work rndv should send both

File: ios/Video/NewPlayerView.swift [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+
+    private func setupDoris() {
+        if let jsBridge = self.jsBridge {
+            let jsProbs = PlayerViewProxy.convertRNVideoJSPropsToRNDV(jsProps: self.jsProps)
+            var jsPlayerView = RNDReactNativeDiceVideo.JSPlayerView(overlayBuilder: JSOverlayBuilder(bridge: jsBridge), jsProps: jsProbs)
+            self.addSubview(jsPlayerView)
+            jsPlayerView.setRCTBubblingEventBlock(onVideoProgress: self.onVideoProgress,
+                                                  onBackButton: self.onBackButton,
+                                                  onVideoError: self.onVideoError,
+                                                  onRequestPlayNextSource: self.onRelatedVideoClicked,
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: also check this one for the same reason as below, 
and double check other callbacks for payload that is expected on JS side

File: ios/Video/NewPlayerView.swift [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+                                                  onSeekToLive: nil,
+                                                  onAdPause: nil,
+                                                  onAdResume: nil,
+                                                  onSubtitleTrackChanged: self.onSubtitleTrackChanged,
+                                                  onAudioTrackChanged: nil,
+                                                  onSeekEvent: nil,
+                                                  onPlaylistEvent: nil,
+                                                  onPlaybackQualityChanged: nil,
+                                                  onShareButton: nil,
+                                                  onRequestHighlightUrl: nil)
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift]: there are 10 callbacks mapped here, please check if all callbacks that are currently used by rnv (global search each) is present in rndv if any is missing will need to add it on rndv side and copy logic from rnv

File: ios/Video/PlayerViewProxy.swift [hunk 20]
```diff
@@ -0,0 +1,279 @@
...
+                    duration: relatedVideo.duration ?? 0.0,
+                    type: relatedVideo.type
+                )
+            }
+            jsPlaylist = JSPlaylist(items: jsPlaylistItems, headIndex: playlist.headIndex)
+        }
+        return jsPlaylist
+    }
+    
+    static func convertRNVideoJSPropsToRNDV(jsProps: JSProps) -> RNDReactNativeDiceVideo.JSProps {
```
Comment (yaroslavlvov) [inline ios/Video/PlayerViewProxy.swift:189]: make all other methods private except this one if they are not used anywhere else

File: ios/Video/NewPlayerView.swift [hunk 21]
```diff
@@ -0,0 +1,194 @@
...
+    @objc var onRequireAdParameters: RCTBubblingEventBlock?
+    @objc var onVideoAboutToEnd: RCTBubblingEventBlock?
+    @objc var onFavouriteButtonClick: RCTBubblingEventBlock?
+    @objc var onRelatedVideoClicked: RCTBubblingEventBlock?
+    @objc var onRelatedVideosIconClicked: RCTBubblingEventBlock?
+    @objc var onStatsIconClick: RCTBubblingEventBlock?
+    @objc var onEpgIconClick: RCTBubblingEventBlock?
+    @objc var onAnnotationsButtonClick: RCTBubblingEventBlock?
+    @objc var onSubtitleTrackChanged: RCTBubblingEventBlock?
+
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift:34]: can you please check old implementation where these callbacks are used and split them here with comment

//used
//not used

we can not just ignore callbacks if they are missing in rndv, if they are missing need to add those in rndv

File: ios/Video/NewPlayerView.swift [hunk 22]
```diff
@@ -141,29 +141,38 @@ class NewPlayerView: UIView, JSInputProtocol {
...
+            let jsPlayerView = RNDReactNativeDiceVideo.JSPlayerView(overlayBuilder: JSOverlayBuilder(bridge: jsBridge), jsProps: jsProbs)
             self.addSubview(jsPlayerView)
+            
+            //set all other props directly here
+            jsPlayerView.onVideoProgress = { [weak self] value in
+                if let currentTime = value?["currentTime"] as? Double {
+                    self?.onVideoProgress?(["currentTime": currentTime])
+                }
+            }
+
```
Comment (yaroslavlvov) [inline ios/Video/NewPlayerView.swift:179]: @NickWang618 set callbacks directly here please

**Review by yaroslavlvov — APPROVED** [review]

**Pull request merged**